fs-storage/src/file_storage.rs

```diff
@@ -280,19 +331,58 @@ mod tests {
 
         let mut file_storage =
             FileStorage::new("TestStorage".to_string(), &storage_path);
+        file_storage.write_fs().unwrap();
+        assert_eq!(file_storage.is_outdated().unwrap(), false);
 
         file_storage.set("key1".to_string(), "value1".to_string());
-        assert_eq!(file_storage.is_storage_updated().unwrap(), false);
+        //todo: we need to add 1ms delays to make the test pass
```

Comment:
@Pushkarm029 could you also complete this test case? The `is_updated` functions wasn't really tested properly.

Reply:
Even after adding delay, it will not pass because `filestorage.set` internally calls `write_fs`, so it no longer remains outdated. I will try to come up with another approach.

Reply:
Perhaps I was too eager in making `set` immediately sync the file system. It'll be more useful to allow the library user to decide whether they want to write to disk immediately after `set`ting a value or use a different strategy like batching writes.

Once `write_fs` is moved out of `set` this test will also be valid.

Reply:
I didn't know this when I refactored the test case, but good that we stumbled upon this. Yes, better to have flexibility regarding when changes are pushed to the disk. In the prototype (arklib-android and Navigator app), we used debounce to write the changes each 1 second or so, because each write was performed within a coroutine and user can create several or more actions per seconds. So, coroutines were racing to write different data into same file, resulting in half-written file... effectively erasing half of the file.

Actually, that was the moment when we realized the need for some protection. With "write into temp file, then move" method, we still can lose some new data (but not everything), with simple mutex we still can overwrite newer data with the older.

Atomic versions came up at this moment, because with them we don't write older versions over the newer versions (mutex + counter + new location every time).